lib/primitives/array-element.js

```diff
@@ -93,7 +96,12 @@ module.exports = function(BaseElement, registry) {
     },
 
     set: function(index, value) {
-      this.content[index] = registry.toElement(value);
+      var item = registry.toElement(value);
+      if (!_.contains(item.parents, this)) {
+        item.parents.push(this);
+      }
+      this.content[index] = item;
+      this.emitSwap();
```

Comment:
Swap is a term used for when new instances and memory allocations are changed in atomic structures. Swap means that a node on the tree has changed. See description here http://omniscientjs.github.io/. 

Note: I want to add more events that can be listened to, such as `update` or `add`.